assignments/tests/test_views.py

```diff
@@ -60,27 +65,101 @@ def test_all_budget_tasks_present_in_section_for_open_assignment(self, create_as
                     break
             assert found, 'Cannot find budgeting task {} in the response'.format(task.name)
 
-        @pytest.mark.django_db
-        def test_all_budget_targets_present_in_budget_tasks_for_open_assignment(self, create_assignments,
-                                                                                assignments_url):
-            assignment = Assignment.objects.filter(status=Assignment.STATUS_OPEN).first()
-            sections = Section.objects.filter(assignment__id=assignment.id)
-            for section in sections:
-                api_client = APIClient()
-                section_url = reverse('assignments:section-detail', args=[assignment.slug, section.id])
-                response = api_client.get(section_url)
-                response_data = response.json()
+    @pytest.mark.django_db
+    def test_all_budget_targets_present_in_budget_tasks_for_open_assignment(self, create_assignments,
+                                                                            assignments_url):
+        assignment = Assignment.objects.filter(status=Assignment.STATUS_OPEN).first()
+        sections = Section.objects.filter(assignment__id=assignment.id)
+        for section in sections:
+            api_client = APIClient()
+            section_url = reverse('assignments:section-detail', args=[assignment.slug, section.id])
+            response = api_client.get(section_url)
+            response_data = response.json()
+
+            for budget_target in BudgetingTarget.objects.filter(budgeting_tasks__section=section):
+                found_target = False
+                for budget_task in budget_target.budgeting_tasks.all():
+                    for data_budget_task in response_data['budgeting_tasks']:
+                        if budget_task.name == data_budget_task['name']:
+                            for data_target in data_budget_task['targets']:
+                                if data_target['name'] == budget_target.name:
+                                    found_target = True
+                                    break
```

Comment:
Gotta take a break.. Too tired to understand where you need 5 nested for loops. Are you sure you need them?